mantis/pyproject.toml

```diff
@@ -11,66 +11,72 @@ packages = [
 ]
 
 [tool.poetry.dependencies]
-python = ">=3.11.6, <=3.11.7"
+python = "3.11.6"
+
+[tool.black]
+line-length = 88
+skip-string-normalization = true
 
-#fastapi-cache2 = "^0.2.1"  # no needed now, too many things
-bech32 = "1.2.0"
-cosmpy = { version = "0.9.1" }
 
-#GLOP, PDLP require ortools which i fail to install
 
-cvxpy = { version = "1.3.2", extras = [
-    "MOSEK",
-    "CBC",
-    "CVXOPT",
-    "GUROBI",
-    "SCIP",
-    "CLARABEL",
-    "GLPK",
-    "XPRESS",
-] }
-ecdsa = "0.18.0"
-fastapi = { extras = ["all"], version = "0.108.0" }
-google-cloud = "0.34.0"
+
+
+#fastapi-cache2 = "^0.2.1"  # no needed now, too many things
+bech32 = "*"
+cosmpy = { version = "0.9.1" }
+cvxpy = { version = "1.3.2", extras = [] }
+ecdsa = "*"
+fastapi = { extras = ["all"], version = "^0.108.0" }
+google-cloud = "*"
 googleapis-common-protos = "1.61.0"
 grpcio = "==1.51.1"
-jsonschema = "4.20.0"
+jsonschema = ">=3.2.0,<5"
 maturin = "1.4.0"
 numpy = "1.23.4"
 pandas = { extras = [], version = "^2.1.4" }
-pendulum = "==2.0.3" # Mac M1 support https://github.com/sdispater/pendulum/issues/454#issuecomment-1847991459
+pendulum = "2.1.2"
```

Comment:
was changed from 2.0.3 made to 2.1.2 made by ruff? i doubt, please update your git head to main head and PR from it.

easier create new fork and create working branch.

try use lazygit